constants/pokemon_data_constants.asm

```diff
@@ -67,39 +69,40 @@ GENDER_UNKNOWN EQU -1
 NUM_DEX_ENTRY_BANKS EQU 4
 
 ; party_struct members (see macros/wram.asm)
-MON_SPECIES            EQUS "(wPartyMon1Species - wPartyMon1)"
-MON_ITEM               EQUS "(wPartyMon1Item - wPartyMon1)"
-MON_MOVES              EQUS "(wPartyMon1Moves - wPartyMon1)"
-MON_ID                 EQUS "(wPartyMon1ID - wPartyMon1)"
-MON_EXP                EQUS "(wPartyMon1Exp - wPartyMon1)"
-MON_STAT_EXP           EQUS "(wPartyMon1StatExp - wPartyMon1)"
-MON_HP_EXP             EQUS "(wPartyMon1HPExp - wPartyMon1)"
-MON_ATK_EXP            EQUS "(wPartyMon1AtkExp - wPartyMon1)"
-MON_DEF_EXP            EQUS "(wPartyMon1DefExp - wPartyMon1)"
-MON_SPD_EXP            EQUS "(wPartyMon1SpdExp - wPartyMon1)"
-MON_SPC_EXP            EQUS "(wPartyMon1SpcExp - wPartyMon1)"
-MON_DVS                EQUS "(wPartyMon1DVs - wPartyMon1)"
-MON_PP                 EQUS "(wPartyMon1PP - wPartyMon1)"
-MON_HAPPINESS          EQUS "(wPartyMon1Happiness - wPartyMon1)"
-MON_PKRUS              EQUS "(wPartyMon1PokerusStatus - wPartyMon1)"
-MON_CAUGHTDATA         EQUS "(wPartyMon1CaughtData - wPartyMon1)"
-MON_CAUGHTLEVEL        EQUS "(wPartyMon1CaughtLevel - wPartyMon1)"
-MON_CAUGHTTIME         EQUS "(wPartyMon1CaughtTime - wPartyMon1)"
-MON_CAUGHTGENDER       EQUS "(wPartyMon1CaughtGender - wPartyMon1)"
-MON_CAUGHTLOCATION     EQUS "(wPartyMon1CaughtLocation - wPartyMon1)"
-MON_LEVEL              EQUS "(wPartyMon1Level - wPartyMon1)"
-MON_STATUS             EQUS "(wPartyMon1Status - wPartyMon1)"
-MON_HP                 EQUS "(wPartyMon1HP - wPartyMon1)"
-MON_MAXHP              EQUS "(wPartyMon1MaxHP - wPartyMon1)"
-MON_ATK                EQUS "(wPartyMon1Attack - wPartyMon1)"
-MON_DEF                EQUS "(wPartyMon1Defense - wPartyMon1)"
-MON_SPD                EQUS "(wPartyMon1Speed - wPartyMon1)"
-MON_SAT                EQUS "(wPartyMon1SpclAtk - wPartyMon1)"
-MON_SDF                EQUS "(wPartyMon1SpclDef - wPartyMon1)"
-BOXMON_STRUCT_LENGTH   EQUS "(wPartyMon1BoxEnd - wPartyMon1)"
-PARTYMON_STRUCT_LENGTH EQUS "(wPartyMon1StructEnd - wPartyMon1)"
-
-NICKNAMED_MON_STRUCT_LENGTH EQUS "(PARTYMON_STRUCT_LENGTH + MON_NAME_LENGTH)"
+rsreset
+MON_SPECIES            rb
+MON_ITEM               rb
+MON_MOVES              rb NUM_MOVES
+MON_ID                 rw
+MON_EXP                rb 3
+MON_STAT_EXP           EQU _RS
+MON_HP_EXP             rw
+MON_ATK_EXP            rw
+MON_DEF_EXP            rw
+MON_SPD_EXP            rw
+MON_SPC_EXP            rw
+MON_DVS                rw
+MON_PP                 rb NUM_MOVES
+MON_HAPPINESS          rb
+MON_PKRUS              rb
+MON_CAUGHTDATA         rw
+MON_CAUGHTLEVEL        EQU _RS - 2
```

Comment:
Doing subtractions on `_RS` feels unclean. Would it be better to use `MON_CAUGHTDATA EQU _RS` to avoid this?

Reply:
There's no exact equivalent of `UNION`s, but there is `RSSET` to "rewind" it.